source/agora/script/ScopeCondition.d

```diff
@@ -0,0 +1,229 @@
+/*******************************************************************************
```

Comment:
In the commit message:

> It's used for tracking which conditional scope are true or not,

It's used for tracking which conditional scopes are true - code may be executed,
or not true - code execution will be skipped until the scope's condition is true again,